src/components/media-player/media-hooks.js

```diff
@@ -23,3 +24,24 @@ export function useMediaDurations(urls) {
   setLastReturnedValue(currentValue);
   return currentValue;
 }
+
+export function useYoutubeThumbnailUrl(url) {
+  const [, setSemaphore] = useState(0);
+  const cache = useService(YoutubeThumbnailUrlCache);
+  const [lastReturnedValue, setLastReturnedValue] = useState(null);
+
+  useEffect(() => {
+    const callback = () => setSemaphore(oldValue => oldValue + 1);
+    cache.subscribe(callback);
+    return () => cache.unsubscribe(callback);
+  }, [cache, setSemaphore]);
```

Comment:
I think `setSemaphore` does not need to be a dependency here, can be removed, right?

Reply:
I was under the impression that we usually had the setters we use in the dependency list, but after scanning through the code I realized that we don't. Removed!